File: Ros_Study/my_message/scripts/talker.py
#!/usr/bin/env python

# -*- coding: utf-8 -*-
# @Author: IBNBlank
# @Date:   2019-01-28 22:50:17
# @Last Modified by:   IBNBlank
# @Last Modified time: 2019-01-29 21:54:21

import rospy
from my_message.msg import chat_msg



if __name__ == '__main__':
    rospy.init_node("talker_py", anonymous=True)
    pub = rospy.Publisher("chatter", chat_msg, queue_size=10)
    rate = rospy.Rate(2)

    count = 0

    while not rospy.is_shutdown():
        msg = chat_msg()
        msg.name = rospy.get_name()
        msg.chat = "hello world {}".format(count)

        rospy.loginfo(msg.chat)
        pub.publish(msg)

        count += 1

        rate.sleep()

File: Ros_Study/my_message/src/talker.cpp
#include <sstream>
#include "ros/ros.h"
#include "ros/this_node.h"
#include "my_message/chat_msg.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "talker_cpp");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<my_message::chat_msg>("chatter", 10);
    ros::Rate loop_rate(2);

    int count = 0;

    while(ros::ok())
    {
        my_message::chat_msg msg;
        std::stringstream ss;

        msg.name = ros::this_node::getName();
        ss << "hello world " << count;
        msg.chat = ss.str();

        ROS_INFO("%s", msg.chat.c_str());
        pub.publish(msg);

        count ++;
        
        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}